packages/core/src/shared/ui/sam/bucketPrompter.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { DefaultS3Client } from '../../clients/s3Client'
+import { samDeployUrl, samSyncUrl } from '../../constants'
+import { createCommonButtons } from '../buttons'
+import { createQuickPick, DataQuickPickItem } from '../pickerPrompter'
+import type { SyncParams } from '../../sam/sync'
+import * as nls from 'vscode-nls'
+import { getRecentResponse } from '../../sam/utils'
+
+const localize = nls.loadMessageBundle()
+export const prefixNewBucketName = (name: string) => `newbucket:${name}`
+
+export enum BucketSource {
```

Comment:
Can we also put header comments to describe the purpose of the function to the reader. I know we haven't done this at most places but given we are refactoring things and placing them in right places to understand the code better, I think it would be a good practice to write a small description.

Reply:
Would you like this done for all function?

Reply:
For the ones that are already in this PR and that too not for all small functions. Maybe describe the purpose where main functionality is happening.

Reply:
I added doc string for exported functions moved to ui/sam files